### ISSUE TITLE
CategoryOrderAttribute Fixes

### ISSUE THREAD

**Issue by MarqueIV** [dotnetprojects/WpfExtendedToolkit#14]
Really pissed at Xceed for changing the licensing of their free library, after treating us like crap for so long.  Even more so pissed as I committed to that library to help them which I would *not* have done had they been charging people.

Speaking of, I created a PR over there for a fix around their `CategoryOrderAttribute` and `PropertyGrid` **_BACK IN AUGUST_**, but they still haven't merged it.  I see your repo is also missing these fixes.

Here's the PR there...

[https://github.com/xceedsoftware/wpftoolkit/pull/1523/commits](https://github.com/xceedsoftware/wpftoolkit/pull/1523/commits)

Not sure how I can point it at yours without downloading and recreating from scratch.  BUT... the time it'll take me to do that, you can probably fix this in five minutes.  It's adding a missing `TypeID` override for the attribute (required when `AllowMultiple` is set to true.  They missed that) and then updating the code to use `TypeDescriptor.GetAttributes` instead of `Type.GetCustomAttributes` as the latter doesn't take into consideration those attributes added at runtime via `TypeDescriptor.AddAttributes`.

Check it out, and hopefully, add it to your code, and I'll make the change from them to yours and I'll be happy to never support them again!

**Comment by MarqueIV** [comment]
Oh... another note, as you can see in the comments [over there](https://github.com/xceedsoftware/wpftoolkit/issues/842), Boucher was wrong about duplicate `CategoryOrderAttribute`s being removed in his example specifically because their code didn't properly set the `TypeID` as my fix does above.  Had they included that, those attributes would *not* have been duplicates, and therefore would not have been removed.

In other words, Boucher's understanding of the problem is wrong because of the bug in his code.

**Comment by jogibear9988** [comment]
Sorry, but I‘ve not really time to look into this, but if you‘d create a pull, I will merge...

**Comment by MarqueIV** [comment]
Will do.  Lemme take a crack at that later today.  Keep an eye out for it.
